component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -858,4 +867,134 @@ private static boolean isURLRelative(String contextFromConfig) throws URISyntaxE
+     */
+    private static Map<String, String> mapFederateClaimsToLocal(ExternalIdPConfig externalIdPConfig, StepConfig stepConfig,
+                                                                AuthenticationContext context)
+            throws TOTPException {
+
+        boolean useDefaultIdpDialect = externalIdPConfig.useDefaultLocalIdpDialect();
+        ApplicationAuthenticator authenticator =
+                stepConfig.getAuthenticatedAutenticator().getApplicationAuthenticator();
+        String idPStandardDialect = authenticator.getClaimDialectURI();
+        Map<ClaimMapping, String> extAttrs = stepConfig.getAuthenticatedUser().getUserAttributes();
+        Map<String, String> originalExternalAttributeValueMap =
+                FrameworkUtils.getClaimMappings(extAttrs, false);
+        Map<String, String> claimMapping = new HashMap<>();
+        Map<String, String> localClaimValues = new HashMap<>();
+        if (useDefaultIdpDialect && StringUtils.isNotBlank(idPStandardDialect)) {
+            try {
+                claimMapping = ClaimMetadataHandler.getInstance()
+                        .getMappingsMapFromOtherDialectToCarbon(idPStandardDialect,
+                                originalExternalAttributeValueMap.keySet(), context.getTenantDomain(),
+                                true);
+            } catch (ClaimMetadataException e) {
+                throw new TOTPException("TOTP Display name creation failed!. Error while handling claim mappings.", e);
+            }
+        } else {
+            ClaimMapping[] customClaimMapping = externalIdPConfig.getClaimMappings();
+            for (ClaimMapping externalClaim : customClaimMapping) {
+                if (originalExternalAttributeValueMap.containsKey(externalClaim.getRemoteClaim().getClaimUri())) {
+                    claimMapping.put(externalClaim.getLocalClaim().getClaimUri(),
+                            externalClaim.getRemoteClaim().getClaimUri());
+                }
+            }
+        }
+
+        if (claimMapping != null && claimMapping.size() > 0) {
+            for (Map.Entry<String, String> entry : claimMapping.entrySet()) {
+                if (originalExternalAttributeValueMap.containsKey(entry.getValue()) &&
+                        originalExternalAttributeValueMap.get(entry.getValue()) != null) {
+                    localClaimValues.put(entry.getKey(),
+                            originalExternalAttributeValueMap.get(entry.getValue()));
+                }
+            }
+        }
+        return localClaimValues;
+    }
+
+    /**
+     * Get the display username for federated users.
+     *
+     * @param context  Authentication context.
+     * @param username Username of the authenticated user.
+     * @return Display username for federated users.
+     * @throws TOTPException When creating the username.
+     */
+    public static String getTOTOIssuerDisplayNameForFederatedUser(AuthenticationContext context, String username)
```

Comment:
Is there any specific reason to make this method public. If not, let's put it as a private method inside TOTPKeyGenerator. WDYT?